pkg/runtime/pubsub/subscriptions.go

```diff
@@ -304,6 +310,9 @@ func marshalSubscription(b []byte) (*Subscription, error) {
 		if err := yaml.Unmarshal(b, &sub); err != nil {
 			return nil, err
 		}
+		if namespace != "" && sub.Namespace != "" && sub.Namespace != namespace {
+			return nil, nil
```

Comment:
Also this. Wonder why this is not covered by tests already written.

Reply:
This is covered by E2E tests (all cases, where `namespace` is empty, `sub.Namespace` is empty and when `sub.Namespace != namespace`)

Reply:
Added

Reply:
The reason why it wasn't already covered was because the test used the v1 CRD but that line was for the v2. I copied the test for v2